### PR TITLE
Enable building native targets for kotest-assertions-json.

### PR DESCRIPTION
**Pull request by charleskorn**
No implementation or documentation changes were required - this was already supported by the current implementation, and the documentation suggested Kotlin/Native was already supported.

Only major change was moving the existing tests into `commonTest` so that they run against all targets.

### CODE REVIEW

File: kotest-assertions/kotest-assertions-json/src/commonTest/kotlin/io.kotest.assertions.json/schema/ArraySchemaTest.kt [hunk 1]
```diff
@@ -3,11 +3,10 @@ package io.kotest.assertions.json.schema
 import io.kotest.assertions.shouldFail
 import io.kotest.core.spec.style.FunSpec
 import io.kotest.matchers.shouldBe
-import org.intellij.lang.annotations.Language
 
 class ArraySchemaTest : FunSpec(
    {
-      fun json(@Language("JSON") raw: String) = raw
+      fun json(raw: String) = raw
```
Comment (Kantis) [inline kotest-assertions/kotest-assertions-json/src/commonTest/kotlin/io.kotest.assertions.json/schema/ArraySchemaTest.kt]: This method and it's usage should be completely removed. It's only point was to provide the language injection 😄

File: kotest-assertions/kotest-assertions-json/src/commonTest/kotlin/io.kotest.assertions.json/schema/ObjectSchemaTest.kt [hunk 2]
```diff
@@ -3,12 +3,11 @@ package io.kotest.assertions.json.schema
 import io.kotest.assertions.shouldFail
 import io.kotest.core.spec.style.FunSpec
 import io.kotest.matchers.shouldBe
-import org.intellij.lang.annotations.Language
 
 
 class ObjectSchemaTest : FunSpec(
    {
-      fun json(@Language("JSON") raw: String) = raw
+      fun json(raw: String) = raw
```
Comment (Kantis) [inline kotest-assertions/kotest-assertions-json/src/commonTest/kotlin/io.kotest.assertions.json/schema/ObjectSchemaTest.kt]: likewise

File: kotest-assertions/kotest-assertions-json/src/commonTest/kotlin/io.kotest.assertions.json/schema/ArraySchemaTest.kt (hunk 1 above)
Comment (charleskorn) [reply kotest-assertions/kotest-assertions-json/src/commonTest/kotlin/io.kotest.assertions.json/schema/ArraySchemaTest.kt]: Good catch - I've fixed this in https://github.com/kotest/kotest/pull/3021/commits/c1dbc548621d9f2dc7cfac13fece6275fd08ae0d.

File: kotest-assertions/kotest-assertions-json/src/commonTest/kotlin/io.kotest.assertions.json/schema/ObjectSchemaTest.kt (hunk 2 above)
Comment (charleskorn) [reply kotest-assertions/kotest-assertions-json/src/commonTest/kotlin/io.kotest.assertions.json/schema/ObjectSchemaTest.kt]: Good catch - I've fixed this in https://github.com/kotest/kotest/pull/3021/commits/c1dbc548621d9f2dc7cfac13fece6275fd08ae0d.

**Pull request merged**